crates/polars-lazy/src/physical_plan/executors/scan/mod.rs

```diff
@@ -114,6 +117,9 @@ impl Executor for AnonymousScanExec {
                 (false, Some(predicate)) => {
                     let mut df = self.function.scan(self.options.clone())?;
                     let s = predicate.evaluate(&df, state)?;
+                    if state.cache_window() {
```

Comment:
This should always evaluate to `false` if this executor didn't activate cached window functions. 

Can we add a `has_window` boolean similar to in `FilterExec` and recreate that behavior?